examples/sktime/flavor.py

```diff
@@ -0,0 +1,639 @@
+    def __init__(self, sktime_model):
+        self.sktime_model = sktime_model
+
+    def predict(self, X):
+        X = None if X.empty else X
+        raw_predictions = {}
+
+        if not hasattr(self.sktime_model, "pyfunc_predict_conf"):
+            raw_predictions[SKTIME_PREDICT] = self.sktime_model.predict(X=X)
+
+        else:
+            if not isinstance(self.sktime_model.pyfunc_predict_conf, dict):
+                raise MlflowException(
+                    f"Attribute {PYFUNC_PREDICT_CONF} must be of type dict.",
+                    error_code=INVALID_PARAMETER_VALUE,
+                )
+
+            if PYFUNC_PREDICT_CONF_KEY not in self.sktime_model.pyfunc_predict_conf:
+                raise MlflowException(
+                    f"Attribute {PYFUNC_PREDICT_CONF} must contain "
+                    f"a dictionary key {PYFUNC_PREDICT_CONF_KEY}.",
+                    error_code=INVALID_PARAMETER_VALUE,
+                )
+
+            if isinstance(self.sktime_model.pyfunc_predict_conf[PYFUNC_PREDICT_CONF_KEY], list):
+                predict_methods = self.sktime_model.pyfunc_predict_conf[PYFUNC_PREDICT_CONF_KEY]
+                predict_params = False
+            elif isinstance(self.sktime_model.pyfunc_predict_conf[PYFUNC_PREDICT_CONF_KEY], dict):
+                predict_methods = list(
+                    self.sktime_model.pyfunc_predict_conf[PYFUNC_PREDICT_CONF_KEY].keys()
+                )
+                predict_params = True
+            else:
+                raise MlflowException(
+                    "Dictionary value must be of type dict or list.",
+                    error_code=INVALID_PARAMETER_VALUE,
+                )
+
+            if not set(predict_methods).issubset(set(SUPPORTED_SKTIME_PREDICT_METHODS)):
+                raise MlflowException(
+                    f"The provided {PYFUNC_PREDICT_CONF_KEY} values must be "
+                    f"a subset of {SUPPORTED_SKTIME_PREDICT_METHODS}",
+                    error_code=INVALID_PARAMETER_VALUE,
+                )
+
+            if SKTIME_PREDICT in predict_methods:
+                raw_predictions[SKTIME_PREDICT] = self.sktime_model.predict(X=X)
+
+            if SKTIME_PREDICT_INTERVAL in predict_methods:
+                if predict_params:
+                    coverage = (
+                        0.9
+                        if "coverage"
+                        not in self.sktime_model.pyfunc_predict_conf[PYFUNC_PREDICT_CONF_KEY][
+                            SKTIME_PREDICT_INTERVAL
+                        ]
+                        else self.sktime_model.pyfunc_predict_conf[PYFUNC_PREDICT_CONF_KEY][
+                            SKTIME_PREDICT_INTERVAL
+                        ]["coverage"]
+                    )
+                else:
+                    coverage = 0.9
+
+                raw_predictions[SKTIME_PREDICT_INTERVAL] = self.sktime_model.predict_interval(
+                    X=X, coverage=coverage
+                )
```

Comment:
can this just be its own method?

Reply:
yes

Reply:
not relevant anymore with revised interface